interfaces/cython/cantera/composite.py

```diff
@@ -108,6 +108,19 @@ class Solution(Transport, Kinetics, ThermoPhase):
     __slots__ = ()
 
 
+class Plasma(Transport, Kinetics, PlasmaPhase):
```

Comment:
I know I already pointed this out earlier, but I'm pretty sure that there are ways to avoid introducing an extra class definition here (it's a bulk phase after all). I'm curious about other opinions (@speth / @bryanwweber)?

Reply:
Maybe cast ThermoPhase to PlasmaPhase in Solution?

Reply:
After research this morning, I still don't know how can I cast the parent class into another after the child class is initiated.

Reply:
Here's an example for a cast from a parent class to a child class: https://github.com/Cantera/cantera/blob/70b048c587a9f7bcbe26458952c02bdd23ab6bc6/interfaces/cython/cantera/reaction.pyx#L312-L313

These are definitely some of the harder-to-understand aspects of Cython :grinning:

Reply:
Okay, I put plasma python function in ThermoPhase and remove PlasmaPhase.

Reply:
Progress! I think you need to make sure that `plasma` related methods aren’t accidentally called for phases that don’t define it as this will result in segfaults.

Reply:
I remember this implementation is similar to IonFlow. The solution was to cast the child class to the parent class after initiation and cast back to the child class when using its member functions. It is a little tedious. Maybe we can figure out better ways to do this. For example, dynamically choose which class Solution(Transport, Kinetics, ThermoPhase/PlasmaPhase).